Integration

strimper13 opened the pull request:
Vision Code up to commit "42be5e"

strimper13 committed:
Vision Code + Network Table

updated vision code and network table put

strimper13 committed:
Add Grippy stuff

strimper13 committed:
Stream Contour

Streams COntours grando

strimper13 committed:
Vision Angles & Brightness v.1

Added Brightness features to allow for tape to be GREEN and begun playing with math for image angles on the contours

strimper13 committed:
Vision Angles v.2

Work in progress towards angle recognition

strimper13 committed:
Found Pairs of rectangles

First Search for the -15 degree contour and add to first part of pair,
#then search for -75 degree contour, put in pair, assure the -75 contour is to the right of the -15 degree contour,
#then assure the -75 contour is the closest contour to the -15,
#by searching for other -75 contours that are either closer or farther

strimper13 committed:
Determines Best Pair of Rectangles

strimper13 committed:
Fixed Network Tables for new code

Updates function to be through -1 to 1 again

msoucy commented:
Entire thing needs to be split into two pull requests - one where you add vision, and one where you add LEDs.

msoucy commented on src/Vision/gripV2.py:
```diff
@@ -0,0 +1,74 @@
+import math
+from enum import Enum
+
+class GripiPipeline:
```
Why is the module named `gripV2` and the class `GripiPipeline`?

strimper13 replied:
Done: Renamed gripipipeline to gripV2

msoucy commented on src/Vision/grippyCode.py:
```diff
@@ -0,0 +1,168 @@
+    leftcX = pair[0][0][0]
+    rightcX = pair[1][0][0]
+    pairMidpoint = (leftcX + rightcX) / 2
+    pairOffset = abs(imageMidpoint - pairMidpoint)
```
Where does `imageMidpoint` come from?

strimper13 replied:
image midpoint is defined in normalizeImage fuction on line 91

msoucy replied:
That's a local variable, that only exists inside `normalizeImage`. You shouldn't be able to access it here.

Please don't leak variables all over the place, it makes kittens sad. 😿

strimper13 replied:
No more sad Kittens! 😿 != 😹

msoucy commented on src/Vision/grippyCode.py:
```diff
@@ -0,0 +1,168 @@
+    for contour in contourList:
+        rectangleBox = cv2.minAreaRect(contour)
+ 
+        if rectangleBox[2] > rightAngle - deadzone and rectangleBox[2] < rightAngle + deadzone:
```
Try `if math.abs(rectangleBox[2] - rightAngle) < deadZone:`

strimper13 replied:
done, will test; commented out old math for now...

strimper13 replied:
Will Attempt to test with ">" instead of "<"

msoucy commented on src/Vision/grippyCode.py:
```diff
@@ -0,0 +1,168 @@
+        print ("rectangle pair coordinates: {}, {}".format(pair[0][0][0],pair[1][0][0]))       
+        if closestCenterPair is None or findPairOffset(pair) < findPairOffset(closestCenterPair):
+            closestCenterPair = pair
+            return closestCenterPair
```
This should probably be outside of the loop, since you only want to return after checking everything.

strimper13 replied:
done; no indents

msoucy commented on src/Vision/grippyCode.py:
```diff
@@ -0,0 +1,168 @@
+cv2.destroyAllWindows()
+
+
+"""
```
Delete this code

strimper13 replied:
deleted line 129

msoucy replied:
I meant to delete the stuff that was all commented out...

msoucy replied:
Should probably re-add that line, and delete everything within `"""`

strimper13 replied:
Fixed
😼

msoucy commented on src/Vision/grippyTest.py:
```diff
@@ -0,0 +1,20 @@
+import cv2
+
+img = cv2.imread('Examples/2.jpg',1)
+cv2.imshow("Plezse show",img)
```
Plezse fix the spelling

strimper13 replied:
always say plezse and tank u

msoucy reviewed (changes requested):
Style change requests - to get these I ran `pycodestyle` on all the code in the folder:

```
.\gripV2.py:6:1: E302 expected 2 blank lines, found 1
.\gripV2.py:9:1: W293 blank line contains whitespace
.\gripV2.py:12:1: W293 blank line contains whitespace
.\gripV2.py:29:5: E303 too many blank lines (2)
.\gripV2.py:35:80: E501 line too long (177 > 79 characters)
.\gripV2.py:39:80: E501 line too long (122 > 79 characters)
.\gripV2.py:42:5: E303 too many blank lines (2)
.\gripV2.py:54:80: E501 line too long (84 > 79 characters)
.\gripV2.py:58:80: E501 line too long (100 > 79 characters)
.\gripV2.py:70:30: E225 missing whitespace around operator
.\gripV2.py:74:1: W391 blank line at end of file
.\grippyCode.py:12:1: E302 expected 2 blank lines, found 1
.\grippyCode.py:18:1: E305 expected 2 blank lines after class or function definition, found 1
.\grippyCode.py:26:11: E231 missing whitespace after ','
.\grippyCode.py:42:1: E265 block comment should start with '# '
.\grippyCode.py:43:1: E265 block comment should start with '# '
.\grippyCode.py:43:80: E501 line too long (115 > 79 characters)
.\grippyCode.py:44:1: E265 block comment should start with '# '
.\grippyCode.py:45:1: E265 block comment should start with '# '
.\grippyCode.py:46:1: E265 block comment should start with '# '
.\grippyCode.py:47:80: E501 line too long (82 > 79 characters)
.\grippyCode.py:48:1: E302 expected 2 blank lines, found 1
.\grippyCode.py:53:5: E265 block comment should start with '# '
.\grippyCode.py:57:1: W293 blank line contains whitespace
.\grippyCode.py:58:80: E501 line too long (95 > 79 characters)
.\grippyCode.py:60:1: W293 blank line contains whitespace
.\grippyCode.py:61:80: E501 line too long (93 > 79 characters)
.\grippyCode.py:62:46: W291 trailing whitespace
.\grippyCode.py:65:25: E231 missing whitespace after ','
.\grippyCode.py:67:13: E265 block comment should start with '# '
.\grippyCode.py:70:37: E231 missing whitespace after ','
.\grippyCode.py:71:20: E711 comparison to None should be 'if cond is not None:'
.\grippyCode.py:76:14: E211 whitespace before '('
.\grippyCode.py:76:73: E231 missing whitespace after ','
.\grippyCode.py:76:80: E501 line too long (88 > 79 characters)
.\grippyCode.py:76:89: W291 trailing whitespace
.\grippyCode.py:77:80: E501 line too long (97 > 79 characters)
.\grippyCode.py:82:1: E265 block comment should start with '# '
.\grippyCode.py:83:1: E265 block comment should start with '# '
.\grippyCode.py:94:1: E305 expected 2 blank lines after class or function definition, found 0
.\grippyCode.py:95:5: E265 block comment should start with '# '
.\grippyCode.py:99:5: E265 block comment should start with '# '
.\grippyCode.py:102:5: E265 block comment should start with '# '
.\grippyCode.py:103:21: E225 missing whitespace around operator
.\grippyCode.py:105:13: E225 missing whitespace around operator
.\grippyCode.py:106:16: E225 missing whitespace around operator
.\grippyCode.py:109:5: E265 block comment should start with '# '
.\grippyCode.py:112:21: E711 comparison to None should be 'if cond is not None:'
.\grippyCode.py:113:80: E501 line too long (132 > 79 characters)
.\grippyCode.py:114:80: E501 line too long (86 > 79 characters)
.\grippyCode.py:117:13: E265 block comment should start with '# '
.\grippyCode.py:117:80: E501 line too long (98 > 79 characters)
.\grippyCode.py:118:1: W293 blank line contains whitespace
.\grippyCode.py:120:5: E265 block comment should start with '# '
.\grippyCode.py:120:5: E303 too many blank lines (2)
.\grippyCode.py:122:5: E265 block comment should start with '# '
.\grippyCode.py:123:23: E231 missing whitespace after ','
.\grippyCode.py:127:1: E303 too many blank lines (3)
.\grippyCode.py:168:1: W391 blank line at end of file
.\grippyTest.py:4:34: E231 missing whitespace after ','
.\grippyTest.py:5:25: E231 missing whitespace after ','
.\grippyTest.py:10:30: E231 missing whitespace after ','
.\grippyTest.py:20:1: W391 blank line at end of file
.\streamTrial.py:6:1: E302 expected 2 blank lines, found 1
.\streamTrial.py:10:1: W293 blank line contains whitespace
.\streamTrial.py:36:5: E303 too many blank lines (2)
.\streamTrial.py:42:80: E501 line too long (177 > 79 characters)
.\streamTrial.py:46:80: E501 line too long (210 > 79 characters)
.\streamTrial.py:50:80: E501 line too long (122 > 79 characters)
.\streamTrial.py:53:5: E303 too many blank lines (2)
.\streamTrial.py:65:80: E501 line too long (84 > 79 characters)
.\streamTrial.py:68:80: E501 line too long (80 > 79 characters)
.\streamTrial.py:79:58: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:79:60: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:79:66: E211 whitespace before '('
.\streamTrial.py:79:80: E225 missing whitespace around operator
.\streamTrial.py:79:80: E501 line too long (84 > 79 characters)
.\streamTrial.py:80:29: E127 continuation line over-indented for visual indent
.\streamTrial.py:80:39: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:80:41: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:80:66: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:80:68: E251 unexpected spaces around keyword / parameter equals
.\streamTrial.py:80:80: E501 line too long (81 > 79 characters)
.\streamTrial.py:84:80: E501 line too long (100 > 79 characters)
.\streamTrial.py:96:35: E225 missing whitespace around operator
.\streamTrial.py:96:80: E501 line too long (83 > 79 characters)
.\streamTrial.py:100:1: W391 blank line at end of file
```

strimper13 committed:
Attempted to fix pull request changes

lbrenn7 committed:
Initial Integration work

This mostly changes the robot maps. It also adds all of the subsystem components to the dashboard

lbrenn7 committed:
Completed Drive Command

Drive command works manually and encoders work, correct angles and distance can be measured

1000DollarBill committed:
Lift Integration

bot190 committed:
More lift integration

Update 3rd party libraries and use brake for controlling the lift

1000DollarBill committed:
Further integration.

-Corrections to Maflipulator logic
-Button mapping
-new "Crappy Lift" command for manual flipping

strimper13 committed:
Now Maflips for real

fixed Alex's code

strimper13 committed:
Prepped Camera for vision & prepped for vision testing

strimper13 committed:
Streams through RoboRIO just badly 

need to get real code on stream

strimper13 committed:
Got Vision Working to the point of align

msoucy committed:
Merge branch 'master' into vision

msoucy commented on src/main/java/frc/robot/subsystems/Manipulator.java:
```diff
@@ -105,7 +115,7 @@ protected boolean isFinished() {
 
             @Override
-            protected void end() {
+            protected void end() {  
```
... There's a space there.

... Why is there a space there?

strimper13 replied:
control shift f is king

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -97,10 +116,10 @@ protected void initialize() {
             protected void usePIDOutput(final double heightCorrection) {
                 double liftSpeed = heightCorrection;
-                if (upperLimit.get() && liftSpeed > 0) {
+                if (!upperLimit.get() && liftSpeed > 0) {
```
Since you had to change all the `upperLimit.get()` to `!upperLimit.get()`, this would be a good place in the future to make a function `boolean isAtUpperLimit() { return !upperLimit.get(); }`. That way you only have to change it in once place.

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -13,29 +19,39 @@
     }
-
-    enum Heights {
+    private void registeredCommands() {
```
Is this used?

strimper13 replied:
BAM 😸

msoucy commented on src/main/java/frc/robot/subsystems/LiftSubsystem.java:
```diff
@@ -13,29 +19,39 @@
     private DigitalInput lowerLimit;
     private DigitalInput upperLimit;
-    private Potentiometer manipAngle;
+    // private CANSparkMax motor;
```
Delete this line if you have the same thing up above

strimper13 replied:
🐈

msoucy commented on src/main/java/frc/robot/RobotMap.java:
```diff
@@ -28,24 +32,20 @@
     ManipulatorMap getManipulatorMap();
 
-    interface ManipulatorMap {
+    public interface ManipulatorMap {
```
You shouldn't need to say that they're public, interface members are public by default.

strimper13 replied:
done-zo 🐩

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -95,28 +110,44 @@ public void teleopInit() {
     }
+
+public void assignButtons() {
+    Robot.xBoxCoPilot.getButton(ButtonXboxController.XBoxButton.BUMPER_LEFT.get()).whenPressed(manipulator.openBeak());
```
Not that it helps the line length much, but you don't need that `.get()` there, you can stop at `BUMPER_LEFT`

msoucy replied:
Also you're already inside `Robot` so you can just say `xBoxCoPilot.getButton(...)`.

strimper13 replied:
done

msoucy commented on .vscode/settings.json:
```diff
@@ -12,5 +12,6 @@
   },
-  "wpilib.online": true
+  "wpilib.online": true,
+  "python.pythonPath": "C:\\Users\\xtron\\AppData\\Local\\Programs\\Python\\Python37-32\\python.exe"
```
Add this to your user settings, not the repo settings file

strimper13 committed:
Fixed Vision multiplier and Matt's comments round two

strimper13 committed:
Actually Fixed Matt's Comments

msoucy commented:
Closed in favor of #16
